src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedSimpleProgramEntryPointSymbol.cs

```diff
@@ -138,7 +140,7 @@ public sealed override bool IsImplicitlyDeclared
         {
             get
             {
-                return true;
+                return false;
```

Comment:
>false [](start = 23, length = 5)

Why is the entry point no longer considered implicitly declared?

Reply:
> Why is the entry point no longer considered implicitly declared?

This is consistent with the way other APIs behave: DeclaringSyntaxReferences, GetDeclaredSymbol, etc. A compilation unit with top level statements declares this method explicitly, there is a node for it.

---
In reply to: [403291559](https://github.com/dotnet/roslyn/pull/43038#discussion_r403291559) [](ancestors = 403291559)

Reply:
The synthesized entry point seems analogous to the synthesized constructor generated for a type with field initializers. The synthesized constructor is marked as implicitly declared.

---
In reply to: [403303550](https://github.com/dotnet/roslyn/pull/43038#discussion_r403303550) [](ancestors = 403303550,403291559)

Reply:
> The synthesized entry point seems analogous to the synthesized constructor generated for a type with field initializers. The synthesized constructor is marked as implicitly declared.

It is not quite analogous. That constructor doesn't have any user code, no syntax belongs to it, it is never an enclosing symbol (GetEnclosingSymbol), there is no way to get semantic model for it and no actions are fired for it, GetDeclaredSymbol doesn't return it, there is no way to get to it other than through GeteMembers on the type.

---
In reply to: [403316449](https://github.com/dotnet/roslyn/pull/43038#discussion_r403316449) [](ancestors = 403316449,403303550,403291559)